src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+    private final Integer maxAnomalyFeatures;
+    private final TimeValue requestTimeout;
+    private final NamedXContentRegistry xContent;
+
+    private ValidateResponse responseValidate;
+    private Map<String, List<String>> failuresMap;
+    private Map<String, List<String>> suggestedChangesMap;
+    private Boolean inferringInterval;
+    private AtomicBoolean inferAgain;
+
+    /**
+     * Constructor function.
+     *
+     * @param client                  ES node client that executes actions on the local node
+     * @param channel                 ES channel used to construct bytes / builder based outputs, and send responses
+     * @param anomalyDetectionIndices anomaly detector index manager
+     * @param anomalyDetector         anomaly detector instance
+     */
+    public ValidateAnomalyDetectorActionHandler(
+        NodeClient client,
+        RestChannel channel,
+        AnomalyDetectionIndices anomalyDetectionIndices,
+        AnomalyDetector anomalyDetector,
+        Integer maxAnomalyDetectors,
+        Integer maxAnomalyFeatures,
+        TimeValue requestTimeout,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(client, channel);
+        this.anomalyDetectionIndices = anomalyDetectionIndices;
+        this.anomalyDetector = anomalyDetector;
+        this.maxAnomalyDetectors = maxAnomalyDetectors;
+        this.maxAnomalyFeatures = maxAnomalyFeatures;
+        this.requestTimeout = requestTimeout;
+        this.responseValidate = new ValidateResponse();
+        this.xContent = xContentRegistry;
+        this.inferringInterval = false;
+        this.inferAgain = new AtomicBoolean(true);
+        this.failuresMap = new HashMap<>();
+        this.suggestedChangesMap = new HashMap<>();
+    }
+
+    /**
+     * Start function to process validate anomaly detector request.
+     * Checks if anomaly detector index exist first, if not, add it as a failure case.
+     *
+     * @throws IOException IOException from {@link AnomalyDetectionIndices#initAnomalyDetectorIndexIfAbsent(ActionListener)}
+     */
+    public void startValidation() throws IOException {
+        if (!anomalyDetectionIndices.doesAnomalyDetectorIndexExist()) {
+            anomalyDetectionIndices
+                .initAnomalyDetectorIndex(
+                    ActionListener.wrap(response -> onCreateMappingsResponse(response), exception -> onFailure(exception))
+                );
+        } else {
+            preDataValidationSteps();
+        }
+    }
+
+    public void preDataValidationSteps() {
+        List<String> missingFields = new ArrayList<>();
+        List<String> formatErrors = new ArrayList<>();
+        if (anomalyDetector.getName() == null || anomalyDetector.getName() == "") {
+            missingFields.add("name");
+        } else if (!anomalyDetector.getName().matches(NAME_REGEX)) {
+            formatErrors.add(anomalyDetector.getName());
+            failuresMap.put("format", formatErrors);
```

Comment:
I would prefer you have a `enum` class for the keys in `failuresMap`, which has `missing`, `others`, `duplicate`, etc, so that this enum can be re-used in anywhere it is used. It would be better to use enum to encapsulate all the types of failures

Reply:
Added enum for failuresMap and suggestedChangesMap